search-resources-api/src/main/java/no/unit/nva/dynamodb/DynamoDBExportFileReader.java

```diff
@@ -0,0 +1,122 @@
+
+
+    public DynamoDBExportFileReader(ElasticSearchHighLevelRestClient elasticSearchRestClient, AmazonS3 s3Client) {
+        this.elasticSearchRestClient = elasticSearchRestClient;
+        this.s3Client = s3Client;
+    }
+
+    private static boolean isPublished(JsonNode jsonNode) {
+        JsonNode statusNode = jsonNode.get(STATUS);
+        if (isNull(statusNode)) {
+            return false;
+        }
+        return statusNode.toString().toLowerCase(Locale.US).contains(PUBLISHED.toLowerCase(Locale.US));
+    }
+
+    /**
+     * Reads one textfile with json dynamodb streamrecords one line at a time.
+     *
+     * @param reader BufferedReader containing json dynamodb records
+     */
+    public void readJsonDataFile(BufferedReader reader) {
+        reader.lines()
+                .map(this::fromJsonString)
+                .filter(Optional::isPresent)
+                .forEach(doc -> addDocumentToIndex(doc.get()));
+
+        logger.info("processed #indexedDocumentCount={}", indexedDocumentCount);
+    }
+
+    /**
+     * Scans an S3 bucket with given key (folder) for files containing dynamodb json data files.
+     *
+     * @param importDataRequest Containing bucket and key for S3
+     * @throws IOException something gone wrong
+     */
+    public void scanS3Folder(ImportDataRequest importDataRequest) {
+
+        ListObjectsV2Result listing =
+                s3Client.listObjectsV2(importDataRequest.getS3bucket(), importDataRequest.getS3folderkey());
+
+        listing.getObjectSummaries()
+                .stream()
+                .filter(this::isDataFile)
+                .map(this::getInputStreamReader)
+                .forEach(this::readJsonDataFile);
+
+    }
+
+    @JacocoGenerated
+    @SuppressWarnings("PMD.CloseResource")
+    private BufferedReader getInputStreamReader(S3ObjectSummary s3ObjectSummary) {
+        GetObjectRequest getObjectRequest =
+                new GetObjectRequest(s3ObjectSummary.getBucketName(), s3ObjectSummary.getKey());
+        S3Object s3Object = s3Client.getObject(getObjectRequest);
+        return new BufferedReader(new InputStreamReader(s3Object.getObjectContent()));
+    }
+
+    private boolean isDataFile(S3ObjectSummary objectSummary) {
+        return objectSummary.getSize() > 0 && !objectSummary.getKey().contains(MANIFEST);
+    }
+
+    private Optional<IndexDocument> fromJsonString(String line) {
+        try {
+            JsonNode node = mapper.readTree(line);
+            if (isPublished(node)) {
+                var indexDocument = IndexDocumentGenerator.fromJsonNode(node);
+                return Optional.of(indexDocument);
+            }
+        } catch (JsonProcessingException e) {
+            logger.error("",e);
+        }
+        return Optional.empty();
+    }
+
+    private void addDocumentToIndex(IndexDocument document) {
+        try {
+            elasticSearchRestClient.addDocumentToIndex(document);
+            indexedDocumentCount++;
```

Comment:
If you return 1 here and 0 to the catch, you can collect the result in the calling method with ".count()" following the map expression calling this method. This removes the need for indexDocumentCount.